pkg/controllers/servicescale/scaler.go

```diff
@@ -302,22 +302,27 @@ func calculateActiveRequests(scanner *bufio.Scanner, requestMatch, responseMatch
 	}
 	var request, response int
 	for scanner.Scan() {
-		response = match(scanner, responseMatch, direction)
-		request = match(scanner, requestMatch, direction)
+		if v, ok := match(scanner.Text(), responseMatch, direction); ok {
+			response = v
+		}
+		if v, ok := match(scanner.Text(), requestMatch, direction); ok {
+			request = v
+		}
```

Comment:
Possible response gets set but request doesn't, and we return a negative ? Same thing I brought up before. Something to think about in future.

Reply:
Don't think there is a chance, but I can do a safe check